### PR TITLE
Added integrating with Amazon S3 topic

### PR DESCRIPTION
**Pull request by gaurav-nelson**
Based on https://help.stackrox.com/docs/integrate-with-other-tools/integrate-with-amazon-s3/

Preview: https://openshift-docs-git-amazons3-gnelson.vercel.app/openshift-acs/master/integration/integrate-with-amazon-s3.html

### CODE REVIEW

File: modules/perform-on-demand-backups-amazon-s3.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Module included in the following assemblies:
+//
+// * integration/integrate-with-amazon-s3.adoc
+:_module-type: PROCEDURE
+[id="perform-on-demand-backups-amazon-s3_{context}"]
+= Performing on demand backups on Amazon S3
```
Comment (gaurav-nelson) [inline modules/perform-on-demand-backups-amazon-s3.adoc:6]: = Performing on-demand backups on Amazon S3

File: integration/integrate-with-amazon-s3.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+* If you are using {product-title} version 3.0.53 or older, the backup does not include certificates.
+* If your Amazon S3 is part of an air-gapped environment, you must add your AWS root CA as a xref:../configuration/add-trusted-ca.adoc[trusted certificate authority] in {product-title}.
+====
+
+include::modules/amazon-s3-configuring-acs.adoc[leveloffset=+1]
+
+include::modules/perform-on-demand-backups-amazon-s3.adoc[leveloffset=+2]
+
+[role="_additional-resources"]
+.Additional resources
```
Comment (bergerhoffer) [inline integration/integrate-with-amazon-s3.adoc:28]: Just checking, is this additional resources section supposed to be for the overall assembly, or smushed with the above module?

If it's for the overall assembly, then it should use the `== Additional resources` format (with an anchor).

File: integration/integrate-with-amazon-s3.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+* xref:../backup_and_restore/backing-up-acs.adoc[Backing up Red Hat Advanced Cluster Security for Kubernetes]
```
Comment (bergerhoffer) [inline integration/integrate-with-amazon-s3.adoc:29]: These xrefs need anchors

File: modules/amazon-s3-configuring-acs.adoc [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// * integration/integrate-with-amazon-s3.adoc
+:_module-type: PROCEDURE
+[id="amazon-s3-configuring-acs_{context}"]
+= Configuring {product-title}
```
Comment (bergerhoffer) [inline modules/amazon-s3-configuring-acs.adoc:6]: This is a pretty generic module title for what it's doing here. Could you possibly make it more specific, as in one of the following? 

* "Configuring an Amazon S3 integration in Red Hat ..." 
* "Integrating Red Hat ... with Amazon S3" 

Actually, as I come back, I'm questioning what this is actually doing. Is it both integrating RHACS with S3, and configuring auto backups? Are auto backups required in order to do the on-demand backups?

So if it's actually setting up auto backups, should it be titled more like "Configuring automatic backups ... "?

And if you don't actually have to set up the auto backups (and could just do the manual on-demand ones), should it be broken up into:

First, setting up the S3 integration, then separate modules for either configuring auto backups or performing an on-demand backup?

File: modules/amazon-s3-configuring-acs.adoc [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+To configure Amazon S3 backups, create a new integration in {product-title}.
+
+.Prerequisites
+* An existing S3 Bucket.
+To create a new bucket with required permissions, see the official Amazon documentation topic link:https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-bucket.html[How Do I Create an S3 Bucket?]
```
Comment (bergerhoffer) [inline modules/amazon-s3-configuring-acs.adoc:12]: When I click the link, the title is just called "Creating a bucket", so I'd use that, and add a period at the end of the sentence.

I think you can also drop "official"

File: modules/amazon-s3-configuring-acs.adoc [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+.Procedure
+. On the RHACS portal, navigate to *Platform Configuration* -> *Integrations*.
+. Scroll down to the *External backups* section and select *Amazon S3*.
+. Click *New Integration* (add icon).
+. Enter a name for *Integration Name*.
+. Enter the number of backups to retain in the *Backups To Retain* box.
+. For *Schedule*, select the backup frequency as daily or weekly and the time to run the backup process.
+. Enter the *Bucket* name in which you want to store the backup.
+. Optionally, enter an *Object Prefix* if you want to save the backups in a specific folder structure.
+For more details, see the official Amazon documentation topic link:https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-keys[Object keys].
```
Comment (bergerhoffer) [inline modules/amazon-s3-configuring-acs.adoc:25]: Same w/ removing "official", and the page name changed if you don't mind updating

File: modules/amazon-s3-configuring-acs.adoc [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+. Enter the *Endpoint* for the bucket if you are using non-public S3 instance, otherwise leave it blank.
```
Comment (bergerhoffer) [inline modules/amazon-s3-configuring-acs.adoc:26]: if you are using -> if you are using a

File: modules/amazon-s3-configuring-acs.adoc [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+* If you are using *KIAM*, *kube2iam* or another proxy, then an *IAM role* that has the `read`, `write`, and `delete` permissions.
+
+.Procedure
+. On the RHACS portal, navigate to *Platform Configuration* -> *Integrations*.
+. Scroll down to the *External backups* section and select *Amazon S3*.
+. Click *New Integration* (add icon).
+. Enter a name for *Integration Name*.
+. Enter the number of backups to retain in the *Backups To Retain* box.
+. For *Schedule*, select the backup frequency as daily or weekly and the time to run the backup process.
+. Enter the *Bucket* name in which you want to store the backup.
```
Comment (bergerhoffer) [inline modules/amazon-s3-configuring-acs.adoc:23]: in which -> where ?

File: modules/perform-on-demand-backups-amazon-s3.adoc [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+Uses the RHACS portal to trigger manual backups of {product-title} on Amazon S3.
+
+.Prerequisites
+* You must have already integrated {product-title} with Amazon S3.
+
+.Procedure
+. On the RHACS portal, navigate to *Platform Configuration* -> *Integrations*.
+. Under the *External backups* section, click *Amazon S3*.
+. Select the integration name for the S3 bucket in which you want to do a backup.
```
Comment (bergerhoffer) [inline modules/perform-on-demand-backups-amazon-s3.adoc:16]: in which -> where ?

**Comment by bergerhoffer** [comment]
This has a conflict

File: modules/amazon-s3-configuring-acs.adoc (hunk 4 above)
Comment (gaurav-nelson) [reply modules/amazon-s3-configuring-acs.adoc:6]: The only reason anyone would integrate ACS with S3 is for automatic backups. Therefore, I'll go with your suggested heading "Configuring Amazon S3 integration in ..."

File: modules/amazon-s3-configuring-acs.adoc (hunk 5 above)
Comment (gaurav-nelson) [reply modules/amazon-s3-configuring-acs.adoc:12]: They might have changed the title from when I last looked at that link :)

Comment (bergerhoffer) [reply modules/amazon-s3-configuring-acs.adoc]: Yeah that's what I assumed :)

**Comment by bergerhoffer** [comment]
Just one little typo, and then a conflict to resolve

**Pull request merged**